pkg/querier/tenantfederation/merge_queryable_test.go

```diff
@@ -84,6 +117,33 @@ func metricMatches(m model.Metric, selector labels.Selector) bool {
 	return selector.Matches(labels.FromStrings(labelStrings...))
 }
 
+type mockSeriesSet struct {
+	upstream storage.SeriesSet
+	warnings storage.Warnings
+	queryErr error
+}
+
+func (m *mockSeriesSet) Next() bool {
+	return m.upstream.Next()
+}
+
+// At returns full series. Returned series should be iteratable even after Next is called.
+func (m *mockSeriesSet) At() storage.Series {
+	return m.upstream.At()
+}
+
+// The error that iteration as failed with.
+// When an error occurs, set cannot continue to iterate.
+func (m *mockSeriesSet) Err() error {
+	return m.queryErr
+}
+
+// A collection of warnings for the whole set.
+// Warnings could be return even iteration has not failed with error.
```

Comment:
```suggestion
// Warnings could be returned even if iteration has not failed with error.
```

Reply:
This acutally seems to be coming from upstream: https://github.com/prometheus/prometheus/pull/8713